test/session.js

```diff
@@ -1258,8 +1257,272 @@ describe('session()', function(){
       });
     });
 
+    it('should set expired cookie to override persistent cookie on req.session = null', function(done){
+      var store = new session.MemoryStore()
+      var app = express()
+        .use(session({ store: store, unset: 'destroy', secret: 'keyboard cat', cookie: { maxAge: min } }))
+        .use(function(req, res, next){
+          req.session.count = req.session.count || 0
+          req.session.count++
+          if (req.session.count === 2) req.session = null
+          res.end()
+        })
+
+      request(app)
+      .get('/')
+      .expect(shouldSetCookie('connect.sid'))
+      .expect(200, function(err, res){
+        if (err) return done(err)
+
+        var now = new Date()
+
+        var expiry = expires(res)
+        assert.ok(expiry)
+
+        var a = new Date(expiry)
+
+        var delta = a.valueOf() - now.valueOf()
+        assert.ok(delta > (min - 1000) && delta <= min)
+
+        request(app)
+        .get('/')
+        .set('Cookie', cookie(res))
+        .expect(shouldSetCookie('connect.sid'))
+        .expect(200, function(err, res){
+          if (err) return done(err)
+
+          now = new Date()
+
+          expiry = expires(res)
+          assert.ok(expiry)
+
+          var b = new Date(expiry)
+
+          delta = b.valueOf() - a.valueOf()
+          assert.ok(delta <= 0)
+
+          delta = b.valueOf() - now.valueOf()
+          assert.ok(delta <= 0)
```

Comment:
Thanks for letting me know.  I'll take a peek and see if/where it makes sense to use.